packages/core/src/scripts/cli.ts

```diff
@@ -1,55 +1,114 @@
+
+    const defaultValue = defaults[key as keyof Flags];
+    // should we default the flag?
+    if (value === undefined) {
+      flags[key as keyof Flags] = defaultValue;
+    }
+
+    if (typeof value !== typeof defaultValue) {
+      throw new Error(`Option '${key}' should be of type ${typeof defaultValue}`);
+    }
+  }
+  return flags as Flags;
+}
 
 export async function cli(cwd: string, argv: string[]) {
   const { input, help, flags } = meow(
     `
     Usage
-      $ keystone [command]
+      $ keystone [command] [options]
     Commands
         dev           start the project in development mode (default)
-        postinstall   generate client APIs and types (optional)
-        build         build the project (must be done before using start)
-        start         start the project in production mode
+        postinstall   build the project (for development, optional)
+        build         build the project (required by \`keystone start\`)
+        start         start the project
         prisma        run Prisma CLI commands safely
         telemetry     sets telemetry preference (enable/disable/status)
+
+    Options
+      --fix (postinstall) @deprecated
+        do build the graphql or prisma schemas, don't validate them
+
+      --frozen (build, prisma)
+        don't build the graphql or prisma schemas, only validate them
+
+      --no-db-push (dev)
+        don't push any updates of your Prisma schema to your database
+
+      --no-prisma (build, dev)
+        don't build or validate the prisma schema
+
+      --no-server (dev)
+        don't start the express server
+
+      --no-ui (build, dev, start)
+        don't build and serve the AdminUI
+
+      --with-migrations (start)
+        trigger prisma to run migrations as part of startup
     `,
     {
-      flags: {
-        fix: { default: false, type: 'boolean' },
-        resetDb: { default: false, type: 'boolean' },
-      },
       argv,
     }
   );
+
   const command = input[0] || 'dev';
-  if (!isCommand(command)) {
-    console.log(`${command} is not a command that keystone accepts`);
-    console.log(help);
-    throw new ExitError(1);
+  if (command === 'dev') {
+    return dev(
+      cwd,
+      defaultFlags(flags, { dbPush: true, prisma: true, server: true, ui: true })
+    );
+  }
+
+  if (command === 'build') {
+    return build(cwd, defaultFlags(flags, { frozen: false, prisma: true, ui: true }));
+  }
+
+  if (command === 'start') {
+    return start(cwd, defaultFlags(flags, { ui: true, withMigrations: false }));
   }
 
   if (command === 'prisma') {
-    return prisma(cwd, argv.slice(1));
-  } else if (command === 'postinstall') {
-    return postinstall(cwd, flags.fix);
-  } else if (command === 'dev') {
-    return dev(cwd, flags.resetDb);
-  } else if (command === 'telemetry') {
-    return telemetry(cwd, argv[1]);
-  } else {
-    return commands[command](cwd);
+    return prisma(cwd, argv.slice(1), defaultFlags(flags, { frozen: false }).frozen);
```

Comment:
The default flags function doesn't work fore when you want to pass through a Prisma flag - for example `--force-reset`